File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>

int main(int argc, string argv[])
{
    string text = get_string("Text: ");
    int letters_count = 0;
    int word_count = 1;
    int sentence_count = 0;

    // count words
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters_count++;
        }
        else if (text[i] == ' ')
        {
            word_count++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence_count++;
        }
    }

    // calculate grade
    float grade = 0.0588 * (100 * letters_count / (float) word_count) - 0.296 * (100 * sentence_count / (float) word_count) - 15.8;
    if (grade < 16 && grade >= 0)
    {
        printf("Grade %i\n", (int) round(grade));
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

File: pset6/readability/readability.py
from sys import exit


"""
Coleman-Liau index
The formula for this is 0.0588 * L - 0.296 * S - 15.8
where
L is the average number of letters per 100 words in the text, and
S is the average number of sentences per 100 words in the text
"""


def coleman_liau_index(l, s):
    return 0.0588 * l - 0.296 * s - 15.8


def main():
    string = input("Text: ")
    letters_count = 0
    word_count = 1
    sentence_count = 0

    for char in string:
        if (char >= "a" and char <= "z") or (char >= "A" and char <= "Z"):
            letters_count += 1
        elif char == " ":
            word_count += 1
        elif char == "." or char == "!" or char == "?":
            sentence_count += 1

    average_letters_per_hundred_words = letters_count / word_count * 100.0
    average_sentences_per_hundred_words = sentence_count / word_count * 100.0
    grade = coleman_liau_index(
        average_letters_per_hundred_words, average_sentences_per_hundred_words)

    if grade < 16 and grade >= 0:
        print(f"Grade {round(grade)}")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print("Before Grade 1")

    exit(0)


main()